test/utils.js

```diff
@@ -81,9 +90,165 @@ async function expectEvent(builder, keypair, eventData = {}) {
+        signers,
+        signatures,
+    }).toBytes();
+
+    let builder = new TxBuilder(client);
+
+    await builder.moveCall({
+        target: `${packageId}::gateway::approve_messages`,
+        arguments: [gateway, messageData, encodedProof],
+    });
+
+    await builder.signAndExecute(keypair);
+
+    builder = new TxBuilder(client);
+
+    const payloadHash = await builder.moveCall({
+        target: `${packageId}::bytes32::new`,
+        arguments: [contractCallInfo.payload_hash],
+    });
+
+    await builder.moveCall({
+        target: `${packageId}::gateway::is_message_approved`,
+        arguments: [
+            gateway,
+            contractCallInfo.source_chain,
+            contractCallInfo.message_id,
+            contractCallInfo.source_address,
+            contractCallInfo.destination_id,
+            payloadHash,
+        ],
+    });
+}
+
+async function approveAndExecuteContractCall(client, keypair, gatewayInfo, messageInfo, executeOptions) {
+    const axelarPackageId = gatewayInfo.packageId;
+    const gateway = gatewayInfo.gateway;
+    const discovery = gatewayInfo.discovery;
+
+    await approveContractCall(client, keypair, gatewayInfo, messageInfo);
+
+    const discoveryArg = [0];
+    discoveryArg.push(...arrayify(discovery));
+    const targetIdArg = [1];
+    targetIdArg.push(...arrayify(messageInfo.destination_id));
+    let moveCalls = [
+        {
+            function: {
+                package_id: axelarPackageId,
+                module_name: 'discovery',
+                name: 'get_transaction',
+            },
+            arguments: [discoveryArg, targetIdArg],
+            type_arguments: [],
+        },
+    ];
+    let isFinal = false;
+
+    while (!isFinal) {
+        const builder = new TxBuilder(client);
+        makeCalls(builder.tx, moveCalls, messageInfo.payload);
+        const resp = await builder.devInspect(keypair.toSuiAddress());
+
+        const txData = resp.results[0].returnValues[0][0];
+        const nextTx = Transaction.parse(new Uint8Array(txData));
+        isFinal = nextTx.is_final;
+        moveCalls = nextTx.move_calls;
+    }
+
+    const builder = new TxBuilder(client);
+    const ApprovedMessage = await builder.moveCall({
+        target: `${axelarPackageId}::gateway::take_approved_message`,
+        arguments: [
+            gateway,
+            messageInfo.source_chain,
+            messageInfo.message_id,
+            messageInfo.source_address,
+            messageInfo.destination_id,
+            messageInfo.payload,
+        ],
+    });
+    makeCalls(builder.tx, moveCalls, messageInfo.payload, ApprovedMessage);
+    return await builder.signAndExecute(keypair, executeOptions);
+}
+
+function makeCalls(tx, moveCalls, payload, ApprovedMessage) {
+    const returns = [];
+
+    for (const call of moveCalls) {
+        let result = tx.moveCall(buildMoveCall(tx, call, payload, ApprovedMessage, returns));
+        if (!Array.isArray(result)) result = [result];
```

Comment:
```suggestion
        if (!Array.isArray(result)) {
            result = [result];
        }
```
avoid single line if statements